FHEM/00_SIGNALduino.pm

```diff
@@ -2071,13 +2073,22 @@ SIGNALduino_Parse_MS($$$$%)
 			$valid = $valid && ($pstr=SIGNALduino_PatternExists($hash,\@{$ProtocolListSIGNALduino{$id}{one}},\%patternList,\$rawData)) >=0;
 			Debug "Found matched one with indexes: ($pstr)" if ($debug && $valid);
 			$patternLookupHash{$pstr}="1" if ($valid); ## Append Sync to our lookuptable
+			$endPatternLookupHash{substr($pstr,0,length($pstr)-1)}="1" if ($valid); ## Append Sync to our lookuptable
+			#SIGNALduino_Log3 $name, 4, "$name: Sync 1 ".$pstr." -> ".substr($pstr,0,length($pstr)-1)." eingetragen"  if ($valid);
 			#Debug "added $pstr " if ($debug && $valid);
 			Debug "one pattern not found" if ($debug && !$valid);
 
 
 			$valid = $valid && ($pstr=SIGNALduino_PatternExists($hash,\@{$ProtocolListSIGNALduino{$id}{zero}},\%patternList,\$rawData)) >=0;
 			Debug "Found matched zero with indexes: ($pstr)" if ($debug && $valid);
 			$patternLookupHash{$pstr}="0" if ($valid); ## Append Sync to our lookuptable
+			if ($valid && !(exists($endPatternLookupHash{substr($pstr,0,length($pstr)-1)}))) {
+				$endPatternLookupHash{substr($pstr,0,length($pstr)-1)}="0" ;
+				#SIGNALduino_Log3 $name, 4, "$name: Sync 0 ".$pstr." -> ".substr($pstr,0,length($pstr)-1)." eingetragen";
+			} else {
+				delete $endPatternLookupHash{substr($pstr,0,length($pstr)-1)};
```

Comment:
wen 0 und 1 die gleiche Sequez bis zum vorletzten haben, dann kann man 0 bzw. 1 nicht bestimmen. Deshalb löschst Du hier den Eintrag.
Habe ich das richtig verstanden`?

Reply:
Genau. Da ich in Perl nicht zu Hause bin hoffe ich, das die Lösung so ok ist. Allerdings kann ich mit dieser Methode den Zweig für float nicht lösen.

Reply:
Alles gut. Um float würde ich mir keine all zu großen Gedanken machen.